lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,142 @@
+require 'uri'
+
+module ProxyAPI
+  class ExternalIpam < ProxyAPI::Resource
+    def initialize(args)
+      @url = args[:url] + "/ipam"
+      super args
+    end
+
+    # Retrieves the next available IP address for the specified subnet
+    #
+    # Inputs: 1. subnet(object): The Foreman subnet
+    #         2. mac: The mac address of the interface obtaining the IP address for
+    # Returns: Hash with "next_ip", or hash with "error"
+    # Examples:
+    #   Response if success:
+    #     {"cidr":"100.20.20.0/24","next_ip":"100.20.20.11"}
+    #   Response if error:
+    #     {"error":"The specified subnet does not exist in external ipam."}
+    def next_ip(subnet, mac)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
```

Comment:
When would `subnet.network` not be present? If it isn't, the `next_ip` call will fail and should rather raise an exception. You can also use the helper
```suggestion
      cidr = subnet.network_address if subnet.network.present?
```